src/ConnectedMode.UnitTests/Suppressions/ServerIssuesStoreTests.cs

```diff
@@ -30,6 +34,88 @@ public class ServerIssuesStoreTests
         public void MefCtor_CheckIsExported()
         {
             MefTestHelpers.CheckTypeCanBeImported<ServerIssuesStore, IServerIssuesStore>();
+            MefTestHelpers.CheckTypeCanBeImported<ServerIssuesStore, IServerIssuesStoreWriter>();
+        }
+
+        [TestMethod]
+        public void AddIssues_ListOfIssuesAsExpected()
+        {
+            var issue1 = CreateIssue("1", false);
+            var issue2 = CreateIssue("2", false);
+            var issue3 = CreateIssue("3", false);
+
+            var testSubject = new ServerIssuesStore();
+
+            var result = testSubject.Get();
+            result.Count().Should().Be(0);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue1, issue2 }, clearAllExistingIssues: false);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(2);
+            result.Should().Contain(issue1);
+            result.Should().Contain(issue2);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue3 }, clearAllExistingIssues: true);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(1);
+            result.Should().Contain(issue3);
+        }
+
+        [TestMethod]
+        public void UpdateIssues_ListOfIssuesAsExpected()
```

Comment:
I'd split this too:
- empty list -> no-op, no event
- non-empty list, no match -> no event
- non-empty list, match -> event, and property changed (if you were being really strict you'd also check that the non-matching items hadn't been changed, only the matching item)